taurus_pyqtgraph/datainspectormodel.py

```diff
@@ -0,0 +1,205 @@
+from datetime import datetime
+
+from pyqtgraph import TextItem, InfiniteLine
+import numpy as np
+from taurustrendset import TaurusTrendSet
+
+
+class DataInspectorModel(object):
```

Comment:
This comment is just an open question since it may be a matter of taste:

In its constructor, `DataInspectorModel` is implicitly "attaching" to the plot item (and effecting changes on the plot item). Likewise, it is implicitly calling `detach` in its destructor. 

I **personally** prefer to be explicit in situations like this. For example, I prefer to explicitly call `detach` instead of simply deleting the object in the `DataInspectorTool._onTriggered` method. The reasons are:
- follow Python Zen of ["explicit better than implicit"](https://www.python.org/dev/peps/pep-0020/)
- It gives the user of the class more control (e.g. the tool could instantiate the inspector just once and attach/detach it instead of instantiating/deleting it each time it is toggled)